### PR TITLE
test: Move test suite into pytest

### PR DESCRIPTION
**Pull request by Swatinem**
This moves unit tests into pytest, in preparation for integration testing

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+def enumerate_unittests():
+    regexp = re.compile("XX\((.*?)\)")
+    tests = []
+    # TODO: actually generate the `tests.inc` file with python
+    with open('tests/unit/tests.inc', 'r') as testsfile:
+        for line in testsfile:
+            match = regexp.match(line)
+            if match:
+                tests.append(match.group(1))
```
Comment (untitaker) [inline tests/conftest.py]: you can use generator functions here: `yield match.group(1)`

File: azure-pipelines.yml [hunk 2]
```diff
@@ -66,11 +62,8 @@ stages:
             fetchDepth: 5
             submodules: recursive
           - script: |
-              mkdir build && cd build
-              cmake ..
-              cmake --build . --parallel --target sentry_tests
-            displayName: Build
-          - script: .\build\sentry_tests.exe
+              pip install --upgrade --requirement integration-test-requirements.txt
```
Comment (jan-auer) [inline azure-pipelines.yml]: Does this also need `UsePythonVersion@0`? Might be worth factoring this out?

File: tests/__init__.py [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+    subprocess.run(configcmd, cwd=cwd, check=True)
+
+    buildcmd = ["cmake", "--build", ".", "--parallel"]
+    for target in targets:
+        buildcmd.extend(["--target", target])
+    subprocess.run(buildcmd, cwd=cwd, check=True)
+
+# Adapted from: https://raw.githubusercontent.com/getsentry/sentry-python/276acae955ee13f7ac3a7728003626eff6d943a8/sentry_sdk/envelope.py
+
+class Envelope(object):
```
Comment (jan-auer) [inline tests/__init__.py]: Let's import this from `sentry_sdk` as soon as it is released.

File: tests/test_integration.c [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+    if (has_arg(argc, argv, "overflow-breadcrumbs")) {
+        for (size_t i = 0; i < 101; i++) {
+            char buffer[4];
+            sprintf(buffer, "%zu", i);
+            sentry_add_breadcrumb(sentry_value_new_breadcrumb(0, buffer));
+        }
+    }
+
+    if (has_arg(argc, argv, "crash")) {
+        memset((char *)0x0, 1, 100);
```
Comment (jan-auer) [inline tests/test_integration.c]: Having this in the example would be really great, too.

File: tests/test_integration.py [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+def assert_meta(envelope):
+    event = envelope.get_event()
+
+    expected = {
+        "platform": "native",
+        "release": "integration-test-release",
+        "user": { "id": 42, "username": "some_name" },
+        "transaction": "test-transaction",
+        "tags": { "expected-tag": "some value" },
+        "extra": { "extra stuff": "some value" },
```
Comment (jan-auer) [inline tests/test_integration.py]: Let's also text `contexts` and `environment`, please.

File: tests/test_integration.c (hunk 4 above)
Comment (Swatinem) [reply tests/test_integration.c]: I actually plan to remove the example completely, and replace it with the integration test file ;-)

**Pull request merged**